### PR TITLE
add context logging

### PR DESCRIPTION
**Pull request by phritz**
i ended up plumbing logging through to the dag and kv because there are some errors logged deep inside that seem worth keeping

progress towards #49 and #176

### CODE REVIEW

File: src/util/rlog/logger.rs [hunk 1]
```diff
@@ -0,0 +1,60 @@
+use log::{debug, error, info};
+
+// Logger is a low-tech context logger. To add context call add_context().
+// To pass it around clone() it. If you can't do that then pass its
```
Comment (aboodman) [inline src/util/rlog/logger.rs:4]: Cloning this just to pass it to a new stack frame seems .. unfortunate. Can we pass by reference typically?

File: src/db/root.rs [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
-pub async fn get_root(store: &dag::Store, head_name: &str) -> Result<String, GetRootError> {
+pub async fn get_root(
+    store: &dag::Store,
+    head_name: &str,
+    logger: rlog::Logger,
+) -> Result<String, GetRootError> {
     use GetRootError::*;
 
-    let read = store.read().await.map_err(ReadError)?;
+    let read = store.read(logger).await.map_err(ReadError)?;
```
Comment (aboodman) [inline src/db/root.rs:11]: Seems like the fact that `read` takes logger by ownership is going to require callers to either clone or not have access to logger after call. Can we pass by ref?

**Review by aboodman — COMMENTED** [review]
Prefer to not establish pattern of all these tiny copies before it's hard to get rid of them. Can we take a pass at attempting to pass by ref?

It might also be easier if we have add_context() use a value pattern, where it returns a copy rather than modifying itself. Not sure.

File: src/util/rlog/logger.rs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+// context() as a String and call new_from_context() to reconstitute it.
```
[the remark this answers is not in the document]
Comment (phritz) [reply src/util/rlog/logger.rs:5]: In dispatch the SENDER is initialized with lazy_static which requires that Request be sync. In the old code I thought it was simpler to just pass the inner String which is itself Sync through the Request than add locking to the LogContext. In the new code this doesn't matter.

File: src/util/rlog/logger.rs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+    pub fn info(&self, msg: String) {
+        info!("{}{}", self.context, msg);
+    }
+
+    pub fn error(&self, msg: String) {
+        error!("{}{}", self.context, msg);
+    }
+
+    // Note: no trace or warn by design!
```
[the remark this answers is not in the document]
Comment (phritz) [reply src/util/rlog/logger.rs:52]: done in https://github.com/rocicorp/repc/pull/199

**Comment by phritz** [comment]
moved to the new scheme in https://github.com/rocicorp/repc/pull/199

**Pull request merged**